src/seedu/addressbook/data/person/Address.java

```diff
@@ -38,22 +45,71 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        return block.toString() + "," + street.toString() + ","
+        		+ unit.toString() + "," + postalCode.toString();
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                && this.toString().equals(((Address) other).toString())); // state check
     }
 
     @Override
     public int hashCode() {
-        return value.hashCode();
+        return this.toString().hashCode();
     }
 
     public boolean isPrivate() {
         return isPrivate;
     }
 }
+
+class Block {
```

Comment:
missing header comments.
how will the users know what this is supposed to do? how will any other developer know what they can do with this class?